server/pruning.go

```diff
@@ -36,3 +43,60 @@ func GetPruningOptionsFromFlags(appOpts types.AppOptions) (pruningtypes.PruningO
 		return pruningtypes.PruningOptions{}, fmt.Errorf("unknown pruning strategy %s", strategy)
 	}
 }
+
+func PruningCmd(providerCreator types.StoreProviderCreator) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "prune",
+		Short: "prune history stetes based on the pruning options specified by flags",
+		Long: `Pruning options can be provided via the '--pruning' flag or alternatively with '--pruning-keep-recent', and
+		'pruning-interval' together.
+		
+		For '--pruning' the options are as follows:
+		
+		default: the last 362880 states are kept, pruning at 10 block intervals
+		nothing: all historic states will be saved, nothing will be deleted (i.e. archiving node)
+		everything: 2 latest states will be kept; pruning at 10 block intervals.
+		custom: allow pruning options to be manually specified through 'pruning-keep-recent', and 'pruning-interval'
+		`,
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			serverCtx := GetServerContextFromCmd(cmd)
+
+			// Bind flags to the Context's Viper so the app construction can set
+			// options accordingly.
+			if err := serverCtx.Viper.BindPFlags(cmd.Flags()); err != nil {
+				return err
+			}
+
+			_, err := GetPruningOptionsFromFlags(serverCtx.Viper)
+			return err
+		},
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			ctx := GetServerContextFromCmd(cmd)
+
+			home := ctx.Viper.GetString(flags.FlagHome)
+
+			db, err := openDB(home, GetAppDBBackend(ctx.Viper))
```

Comment:
https://github.com/cosmos/cosmos-sdk/pull/11361/files#diff-e6374d17b47e21bb322f908dc81b6a544d9c57ca2569b1e9b28572af68551963R33
You may need the `appCreator` to initialize the multistore properly.

Reply:
I specify `appCreator` as `providerCreator`, which is an instance of `StoreProviderCreator` interface:
```go
StoreProviderCreator func(log.Logger, dbm.DB, AppOptions) RootMultiStoreProvider
```